### DISCUSSION TITLE
Passing referer option to download a file

### DISCUSSION THREAD

**Discussion by cesarcoatl** [ScoopInstaller/Scoop#5140] [Help and Questions] [answered]
Hello,

I am trying to write a manifest for an application where the download requires a referer. On the command line I can use the `--referer` option for `curl` or `wget`.

I've read the App Manifests docs, but failed to find that option.

Thanks

**Comment by cesarcoatl** [comment]
I am thinking if I could download the file using `Invoke-WebRequest` and passing Referer manually in the Header at `pre_install`.

Thoughts?

**Comment by rashil2000** [comment, marked as answer]
We don't directly support the `Referer` option, but we have a feature to allow custom headers - can you see if that works for you? Ref: https://github.com/ScoopInstaller/Scoop/pull/4254#issuecomment-1068514131

**Reply by cesarcoatl** [reply]
That did it!

I had to manually add that section on my configuration file (`%USERPROFILE%\.config\scoop\config.json`) and it now looks like this:

```json
{
  "lastUpdate": "2022-09-09T07:21:08.0363143-07:00",
  "SCOOP_REPO": "https://github.com/ScoopInstaller/Scoop",
  "SCOOP_BRANCH": "master",
  "private_hosts": [
    {
      "match": "https://files.domain.com/*",
      "headers": "Referer=https://domain.com/"
    }
  ]
}

```

Is it even possible to configure that using the `scoop config private_hosts` command?

In Hombrew, where the `curl` download strategy is the default, one can pass `referer` as part of the `url`. Would you consider adding this in the future to the app manifest definition?

Thanks

**Reply by rashil2000** [reply]
> Is it even possible to configure that using the `scoop config private_hosts` command?

Not currently, you need to edit the config manually.


> In Hombrew, where the `curl` download strategy is the default, one can pass `referer` as part of the `url`. Would you consider adding this in the future to the app manifest definition?

It is possible, but the schema would soon become cluttered because there are lots of types of headers that we would want to support. This is why we have the more flexible option (as described above) currently.

**Reply by cesarcoatl** [reply]
Thanks, @rashil2000!

**Comment by sinloss** [comment]
Is there any update on this? How much control do we now have on the downloading requests' headers?

Asking for #6009
